### DISCUSSION TITLE
Validate dynamic page param

### DISCUSSION THREAD

**Discussion by kevbot-git** [nuxt/nuxt#17889] [Questions]
I'm trying to do some validation on the slug for a route within the page, for example:
`pages/person/[name].vue`
```vue
<script setup>
// ...within some sort of validation hook
  return ['fred', 'jenny'].includes(params.name);
</script>
```
...at which point I'd want to redirect to a 404 page if the name doesn't validate.
Is there an equivalent method to this one from Nuxt 2?
https://nuxtjs.org/docs/components-glossary/validate/

**Comment by fabianwohlfart** [comment]
Did you find and answer?

**Comment by lewebsimple** [comment]
You can achieve this with a [middleware](https://v3.nuxtjs.org/docs/directory-structure/middleware#middleware-directory)
